addons/hitreactions/functions/fnc_checkWeaponDrop.sqf

```diff
@@ -0,0 +1,37 @@
+#include "..\script_component.hpp"
+/*
+ * Author: KJW
+ * Checks if unit should drop weapon based on projectile hit info
+ *
+ * Arguments:
+ * 0: surfaceType <STRING>
+ * 1: components <ARRAY>
+ * 2: position <ARRAY>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * ["", [], [1,1,1]] call ACE_hitreactions_fnc_checkWeaponDrop
+ *
+ * Public: No
+ */
+
+params ["_surfaceType", "_components", "_position"];
+
+if (!(_surfaceType isEqualTo "") && !(_components findAny GVAR(armComponents) isEqualTo -1)) exitWith {}; // Could be potential for edge cases here however couldn't find any from testing.
```

Comment:
Alternatively, since it exists (and a huge group of ppl dunno about it):
```suggestion
if (_surfaceType isNotEqualTo "" && (_components findAny GVAR(armComponents) isNotEqualTo -1)) exitWith {}; // Could be potential for edge cases here however couldn't find any from testing.
```